jmespath-jackson-jr/src/main/java/io/burt/jmespath/jacksonjr/JacksonJrRuntime.java

```diff
@@ -0,0 +1,225 @@
+                .build());
+    }
+    public JacksonJrRuntime(RuntimeConfiguration configuration, JSON json) {
+        super(configuration);
+        this.json = json;
+    }
+
+    @Override
+    public TreeNode parseString(String str) {
+        try {
+            return json.treeFrom(str);
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static class JrsArrayListWrapper extends AbstractList<TreeNode> {
+        private final JrsArray array;
+
+        JrsArrayListWrapper(JrsArray array) {
+            this.array = array;
+        }
+
+        @Override
+        public TreeNode get(int index) {
+            return array.get(index);
+        }
+
+        @Override
+        public int size() {
+            return array.size();
+        }
+    }
+
+    @Override
+    public List<TreeNode> toList(TreeNode value) {
+        if (value == null) {
+            return Collections.emptyList();
+        }
+        if (value.isArray()) {
+            return new JrsArrayListWrapper((JrsArray) value);
+        } else if (value.isObject()) {
+            JrsObject object = (JrsObject) value;
+            List<TreeNode> list = new ArrayList<>(object.size());
+            Iterator<Map.Entry<String, JrsValue>> iterator = object.fields();
+
+            while (iterator.hasNext()) {
+                Map.Entry<String, TreeNode> entry = (Map.Entry) iterator.next();
+                list.add(entry.getValue());
+            }
+            return list;
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String toString(TreeNode value) {
+        if (value.asToken().equals(JsonToken.VALUE_STRING)) {
+            return ((JrsString) value).asText();
+        } else {
+            try {
+                return json.asString(value);
+            } catch (IOException e) {
+                return "";
+            }
+        }
+    }
+
+    @Override
+    public Number toNumber(TreeNode value) {
+        if (value.isValueNode() && ((JrsValue) value).isNumber()) {
+            JrsNumber number = (JrsNumber) value;
+            return number.getValue();
+        } else return null;
+    }
+
+    @Override
+    public boolean isTruthy(TreeNode value) {
+        // false, null, empty lists, empty objects, empty strings.
+        if (value.isContainerNode()) {
+            return value.size() > 0;
+        } else if (value.isValueNode()) {
+            if (value.asToken().equals(JsonToken.VALUE_STRING)) {
+                return !((JrsString) value).asText().isEmpty();
+            } else return !value.asToken().equals(JsonToken.VALUE_FALSE) &&
+                    !value.asToken().equals(JsonToken.VALUE_NULL);
+        } else {
+            return !value.isMissingNode();
+        }
+    }
+
+    @Override
+    public JmesPathType typeOf(TreeNode value) {
+        switch (value.asToken()) {
+            case START_ARRAY:
+            case END_ARRAY:
```

Comment:
Can an `END_ARRAY` token appear here at all? What `TreeNode` will get that token? (same with `END_OBJECT`).